src/NHibernate/Criterion/InExpression.cs

```diff
@@ -62,41 +58,50 @@ public override SqlString ToSqlString(ICriteria criteria, ICriteriaQuery criteri
 				return new SqlString("1=0");
 			}
 
-			//TODO: add default capacity
-			SqlStringBuilder result = new SqlStringBuilder();
-			SqlString[] columnNames =
-				CriterionUtil.GetColumnNames(_propertyName, _projection, criteriaQuery, criteria);
-			
-			// Generate SqlString of the form:
-			// columnName1 in (values) and columnName2 in (values) and ...
```

Comment:
This is wrong  as instead of  requested (1, 1), (2, 2), (3, 3) values it will find all combinations of those values as (1,2), (1,3), (2,1)...